google-cloud-storage/lib/google/cloud/storage/hmac_key.rb

```diff
@@ -0,0 +1,312 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/storage/hmac_key/list"
+
+module Google
+  module Cloud
+    module Storage
+      ##
+      # # HmacKey
+      #
+      # Represents the metadata for an HMAC key, and also includes the key's
+      # secret if returned by the {Project#create_hmac_key} creation method.
+      #
+      # @see https://cloud.google.com/storage/docs/migrating#migration-simple
+      #   Migrating from Amazon S3 to Cloud Storage - Simple migration
+      #
+      # @attr_reader [String, nil] secret HMAC secret key material, or `nil` if
+      #   the key was not returned by the {Project#create_hmac_key} creation
+      #   method.
+      #
+      # @example
+      #   require "google/cloud/storage"
+      #
+      #   storage = Google::Cloud::Storage.new
+      #
+      #   service_account_email = "my_account@developer.gserviceaccount.com"
+      #   hmac_key = storage.create_hmac_key service_account_email
+      #   hmac_key.secret # ...
+      #
+      #   hmac_key = storage.hmac_key hmac_key.access_id
+      #   hmac_key.secret # nil
+      #
+      #   hmac_key.inactive!
+      #   hmac_key.delete
+      #
+      class HmacKey
+        ##
+        # @private The Connection object.
+        attr_accessor :service
+
+        ##
+        # @private The Google API Client object.
+        attr_accessor :gapi
+
+        ##
+        # @private A boolean value or a project ID string to indicate the
+        # project to be billed for operations.
+        attr_accessor :user_project
+
+        attr_reader :secret
+
+        ##
+        # @private Creates a Notification object.
```

Comment:
Nit: Use HmacKey instead of Notification object.

Reply:
Fixed.